python/pysrc/kaskada/_timestream.py

```diff
@@ -1463,9 +1463,15 @@ def _aggregation(
     if window is None:
         return Timestream._call(op, input, *args, None, None)
     elif isinstance(window, kd.windows.Since):
-        return Timestream._call(op, input, *args, window.predicate, None)
+        predicate = window.predicate
+        if callable(predicate):
+            predicate = predicate(input)
+        return Timestream._call(op, input, *args, predicate, None)
     elif isinstance(window, kd.windows.Sliding):
-        return Timestream._call(op, input, *args, window.predicate, window.duration)
+        predicate = window.predicate
+        if callable(predicate):
+            predicate = predicate(input)
+        return Timestream._call(op, input, *args, predicate, window.duration)
```

Comment:
We'll probably have to do something similar to this if we want to support ticks in other functions like `filter/if`. I could work on that as a follow-up to this